lib/ae_mdw_web/plugs/paginated_plug.ex

```diff
@@ -148,40 +137,87 @@ defmodule AeMdwWeb.Plugs.PaginatedPlug do
     do: {:ok, :backward, @default_scope}
 
   defp extract_direction_and_scope(%{"direction" => direction}, _txi_scope?, _state),
-    do: {:error, "invalid direction: #{direction}"}
+    do: {:error, ErrInput.Query.exception(value: "invalid direction `#{direction}`")}
 
   defp extract_direction_and_scope(_params, _txi_scope?, _state),
     do: {:ok, :backward, @default_scope}
 
-  defp extract_range(range) when is_binary(range) do
+  defp extract_range(state, scope_type, range, params) when is_binary(range) do
     case String.split(range, "-") do
       [from, to] ->
         case {Integer.parse(from), Integer.parse(to)} do
-          {{from, ""}, {to, ""}} when from >= 0 and to >= 0 -> {:ok, from, to}
-          {_from_err, _to_err} -> {:error, "invalid range: #{range}"}
+          {{from, ""}, {to, ""}} when from >= 0 and to >= 0 ->
+            generate_range(state, scope_type, from, to, params)
+
+          {_from_err, _to_err} ->
+            {:error, ErrInput.Scope.exception(value: range)}
         end
 
       [single] ->
         case Integer.parse(single) do
-          {single, ""} when single >= 0 -> {:ok, single, single}
-          _single_err -> {:error, "invalid range: #{range}"}
+          {single, ""} when single >= 0 ->
+            generate_range(state, scope_type, single, single, params)
+
+          _single_err ->
+            {:error, ErrInput.Scope.exception(value: range)}
         end
 
       _splitted_range ->
-        {:error, "invalid range: #{range}"}
+        {:error, ErrInput.Scope.exception(value: range)}
+    end
+  end
+
+  defp extract_range(_state, _scope_type, range, _params),
+    do: {:error, ErrInput.Scope.exception(value: "invalid range: #{inspect(range)}")}
+
+  defp generate_range(state, :time, first, last, params) do
+    with {_first, {:ok, first_parsed}} <- {first, DateTime.from_unix(first)},
+         {_last, {:ok, last_parsed}} <- {last, DateTime.from_unix(last)} do
+      {first_txi, last_txi} =
+        DbUtil.time_to_txi(
+          state,
+          DateTime.to_unix(first_parsed, :millisecond),
+          DateTime.to_unix(last_parsed, :millisecond)
+        )
+
+      generate_range(state, :txi, first_txi, last_txi, params)
+    else
+      {invalid_unix_time, {:error, _reason}} ->
+        {:error, ErrInput.Scope.exception(value: "invalid unix time `#{invalid_unix_time}`")}
     end
   end
 
-  defp extract_range(range), do: {:error, "invalid range: #{range}"}
+  defp generate_range(_state, scope_type, first, last, _params) when first < last do
+    {:ok, :forward, {scope_type, first..last}}
+  end
+
+  defp generate_range(_state, scope_type, first, last, _params) when last < first do
+    {:ok, :backward, {scope_type, last..first}}
+  end
+
+  defp generate_range(_state, scope_type, first, last, params) do
```

Comment:
since this is the match for `first = last`, maybe instead of calling `first` and `last` just call it smth like `gen`? 
It made me thinking about it after looking at the
```
      {:ok, :forward, {scope_type, last..first}}
```
while above we have
```
    {:ok, :forward, {scope_type, first..last}}
```